ISSUER_PROTOCOL.md

```diff
@@ -0,0 +1,333 @@
+                     "expiry": <key expiry, encoded as a string representation of
+                                an integer timestamp in microseconds since the Unix
+                                epoch> },
+  <keyID>: { "Y": <base64-encoded public key>,
+                     "expiry": <key expiry, encoded as a string representation of
+                                an integer timestamp in microseconds since the Unix
+                                epoch> }, ...
+}
+```
+
+
+### Issuing Tokens
+
+To support the issuance of tokens made via Trust Token calls by the client, server paths that support token issuance should parse the "Sec-Trust-Token" header as a base64 binary blob. This decoded binary blob should be interpreted as a **Trust Token Issuance Request** and passed into the _Issue_ crypto protocol along with the Issuance metadata based on the rest of the request. The result of _Issue_ (a **Trust Token Issuance Response**) should be base64 encoded and returned via the "Sec-Trust-Token" header in the HTTP response.
+
+
+#### Issuance Metadata
+
+As part of an issuance, two forms of metadata can be embedded into the token. Public metadata is embedded via the choice of key which is used as part of the issuance process (amongst the up to 3 keys configured in the key commitment) and is passed into the _Issue_ method as the keypair selection. Private metadata is embedded via a cryptographically hidden bit in the signed token itself and is passed into the _Issue_ method as the private metadata boolean.
+
+
+### Redeeming Tokens
+
+To support the redemption of tokens by the client, server paths that support token redemption should parse the "Sec-Trust-Token" header as a base64 binary blob. This decoded binary blob should be interpreted as a **Trust Token Redemption Request** and passed into the _Redeem_ crypto protocol. The result of _Redeem_ (a **Trust Token Redemption Response**) should be base64 encoded and returned via the "Sec-Trust-Token" header in the HTTP response. Additionally, the issuer should keep track of redeemed tokens to prevent token reuse from malicious clients.
+
+
+#### Redemption Metadata
+
+At redemption time, the metadata encoded in the Trust Token will be embedded in the Signed Redemption Record. The public metadata is embedded via the key ID of the corresponding key used to sign the Trust Token. The private metadata bit is embedded as a one-bit boolean that can be set per redeemer logic and which can be decoded by partners downstream. One potential means of encoding the private metadata is to hash a shared secret and the token hash and take the lowest bit XOR between that and the metadata value (H(shared\_secret||token-hash) ^ private metadata value), allowing only partners with the shared secret to get the true value of the private metadata bit. Other potential schemes involve an additional endpoint which can be used to read the private metadata value based on the SRR or token hash.
```

Comment:
Suggestion: link to the definition of SRR in the explainer (or design doc) so that readers who don't know what an SRR is can see it

Reply:
Expanded to the full name.